src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -199,61 +204,49 @@ func (h *ConnectIntegrationHandler) Perform(ctx context.Context, interfaceArgs i
 	}
 
 	if args.SetAsStorage {
-		// This integration should be used as the new storage layer.
-		// In order to do so, we need to migrate all content from the old store
-		// to the new store. This requires pausing the server and then restarting it.
-		// All of this logic is performed asynchronously so that the user knows that
-		// the connect integration request has succeeded and that the migration is now
-		// under way.
-		go func() {
-			log.Info("Starting storage migration process...")
-			// Wait until the server is paused
-			h.PauseServer()
-			// Makes sure that the server is restarted
-			defer h.RestartServer()
-
-			// Wait until there are no more workflow runs in progress
-			lock := utils.NewExecutionLock()
-			if err := lock.Lock(); err != nil {
-				log.Errorf("Unexpected error when acquiring workflow execution lock: %v. Aborting storage migration!", err)
-				return
-			}
-			defer func() {
-				if err := lock.Unlock(); err != nil {
-					log.Errorf("Unexpected error when unlocking workflow execution lock: %v", err)
-				}
-			}()
+		confData, err := args.Config.Marshal()
+		if err != nil {
+			return nil, http.StatusInternalServerError, err
+		}
 
-			if err := setIntegrationAsStorage(
-				context.Background(),
-				args.Service,
-				args.Config,
-				args.OrgID,
-				h.DAGRepo,
-				h.ArtifactRepo,
-				h.ArtifactResultRepo,
-				h.OperatorRepo,
-				h.IntegrationRepo,
-				h.Database,
-			); err != nil {
-				log.Errorf("Unexpected error when setting the new storage layer: %v", err)
-			}
+		newStorageConfig, err := storage.ConvertIntegrationConfigToStorageConfig(args.Service, confData)
+		if err != nil {
+			return emptyResp, http.StatusBadRequest, errors.Wrap(err, "Integration config is malformed.")
+		}
 
-			log.Info("Successfully migrated the storage layer!")
-		}()
+		err = storage_migration.PerformStorageMigration(
```

Comment:
@saurav-c my understanding is we don't need to name method repeatedly after package name? In this case, would `Perform()` be sufficient here?

Reply:
Yeah, that would be good practice